### PR TITLE
Fix pvc cache

### PR DESCRIPTION
**Pull request by horizen**
fix issue #1983

### CODE REVIEW

File: pkg/scheduler/cache/cache.go [hunk 1]
```diff
@@ -701,19 +725,9 @@ func (sc *SchedulerCache) BindPodGroup(job *schedulingapi.JobInfo, cluster strin
...
-// AllocateVolumes allocates volume on the host to the task
-func (sc *SchedulerCache) AllocateVolumes(task *schedulingapi.TaskInfo, hostname string, podVolumes *volumescheduling.PodVolumes) error {
-	return sc.VolumeBinder.AllocateVolumes(task, hostname, podVolumes)
-}
-
-// BindVolumes binds volumes to the task
-func (sc *SchedulerCache) BindVolumes(task *schedulingapi.TaskInfo, podVolumes *volumescheduling.PodVolumes) error {
-	return sc.VolumeBinder.BindVolumes(task, podVolumes)
+// SchedulerVolumeBinder returns volume binder
+func (sc *SchedulerCache) SchedulerVolumeBinder() VolumeBinder {
```
Comment (k82cn) [inline pkg/scheduler/cache/cache.go:729]: Let's just add a `RevertVolumes` func for cache.

Comment (k82cn) [reply pkg/scheduler/cache/cache.go:729]: further more, I'd like improve the steps as follows:

1. allocate remote resources (volumen, remote gpu pool...)
2. allocate local resource (cpu, mem, gpu ...)
3. discard if any error

Which means we should not enhance this case by case :) Anyway, I'm ok to get this merged firstly to handle the issue, and open a seperate PR to do the refactor.

Comment (horizen) [reply pkg/scheduler/cache/cache.go:729]: > Let's just add a `RevertVolumes` func for cache.

My thought is cache‘s get/allocate/bind method is just a proxy for VolumeBinder and VolumeBinder is also public interface, returns volumebinder may simplify the code. 

I will update it for that.

File: pkg/scheduler/cache/cache.go [hunk 2]
```diff
@@ -275,15 +275,38 @@ func (dvb *defaultVolumeBinder) AllocateVolumes(task *schedulingapi.TaskInfo, ho
 	return err
 }
 
+// AllocateVolumes allocates volume on the host to the task
```
Comment (Thor-wl) [inline pkg/scheduler/cache/cache.go:278]: The function name is not match the reality.

Comment (horizen) [reply pkg/scheduler/cache/cache.go]: > The function name is not match the reality.

update

File: pkg/scheduler/cache/cache.go (hunk 1 above)
Comment (horizen) [reply pkg/scheduler/cache/cache.go]: > Let's just add a `RevertVolumes` func for cache.

updated

File: pkg/scheduler/util/test_utils.go [hunk 3]
```diff
@@ -93,6 +97,102 @@ func BuildPod(namespace, name, nodename string, p v1.PodPhase, req v1.ResourceLi
...
+			NodeSelector: selector,
+			Containers: []v1.Container{
+				{
+					Resources: v1.ResourceRequirements{
+						Requests: req,
+					},
+					VolumeMounts: []v1.VolumeMount{
+						{
+							Name:      pvc.Name,
+							MountPath: "/data",
```
Comment (Thor-wl) [inline pkg/scheduler/util/test_utils.go:126]: Fixed mount path?

**Comment by Thor-wl** [comment]
Pls fix the DCO. And I'm generally OK about this PR.
/lgtm

**Comment by horizen** [comment]
seems like go mod outdate

**Comment by Thor-wl** [comment]
> seems like go mod outdate

Yes, please rebase the latest code of master branch.

**Review by hwdef — COMMENTED** [review]
please update go.mod and vendor folder

**Comment by horizen** [comment]
> please update go.mod and vendor folder

just rebase the latest change of master

**Pull request merged**